### ISSUE TITLE
Modal component caused Invariant Violation: Maximum update depth exceeded with react-native 0.50.3

### ISSUE THREAD

**Issue by shcheuk** [callstack/react-native-paper#216]
To reproduce,
```
import * as React from 'react';
import { AppRegistry } from 'react-native';
import {Modal, Provider as PaperProvider } from 'react-native-paper';
import App from './src/App';

function Main() {
  return (
    <PaperProvider>
      <Modal visible={true}/>
    </PaperProvider>
  );
}

AppRegistry.registerComponent('main', () => Main);
```
Error message received:
`Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`

if I remove the <ThemedPortal> in modal.js render function , it will works fine. It seems the <ThemedPortal> make it keep rendeing

**Comment by shcheuk** [comment]
same issue also happen on dialog as dialog use modal
temporary remove <ThemedPortal> in Modal.js render element will make the error disappear, so the issue seems come from theme portal

**Comment by satya164** [comment]
I'll check it tomorrow

**Comment by satya164** [comment]
Hey! I sent a PR which should fix it - #261 

Will be awesome if you try it out and let me know if it works.
